### PR TITLE
update categorical profiler

### PR DESCRIPTION
**Pull request by atl1502**

### CODE REVIEW

File: dataprofiler/profilers/categorical_column_profile.py [hunk 1]
```diff
@@ -678,6 +681,9 @@ def update(self, df_series: Series) -> CategoricalColumn:
         :return: updated CategoricalColumn
         :rtype: CategoricalColumn
         """
+        # TODO remove onces profiler builder is updated
+        if type(df_series) == pd.Series:
```
Comment (taylorfturner) [inline dataprofiler/profilers/categorical_column_profile.py:685]:
```suggestion
        if isinstance(df_series, pd.Series):
```

Comment (taylorfturner) [reply dataprofiler/profilers/categorical_column_profile.py:685]: you'll want to validate but this is a bit cleaner

File: dataprofiler/tests/profilers/test_categorical_column_profile.py [hunk 2]
```diff
@@ -19,6 +20,8 @@
 
 
 class TestCategoricalColumn(unittest.TestCase):
+    maxDiff = None
+
```
Comment (taylorfturner) [inline dataprofiler/tests/profilers/test_categorical_column_profile.py:24]: def remove

File: dataprofiler/tests/profilers/test_categorical_column_profile.py [hunk 3]
```diff
@@ -883,7 +888,8 @@ def test_json_encode(self):
         self.assertEqual(serialized, expected)
 
     def test_json_encode_after_update(self):
-        df_categorical = pd.Series(
+        df_categorical = pl.Series(
+            None,
```
Comment (taylorfturner) [inline dataprofiler/tests/profilers/test_categorical_column_profile.py:892]: why `None` outside `[]`?

**Review by taylorfturner — CHANGES_REQUESTED** [review]

**Review by taylorfturner — DISMISSED** [review]

File: dataprofiler/profilers/categorical_column_profile.py [hunk 4]
```diff
@@ -601,7 +603,8 @@ def _get_categories_full(self, df_series) -> dict:
         :return: dict of counts for each unique value
         :rtype: dict
         """
-        category_count: dict = df_series.value_counts(dropna=False).to_dict()
+        value_counts = df_series.value_counts(sort=True)
+        category_count: dict = dict(value_counts.iter_rows())
```
Comment (micdavis) [inline dataprofiler/profilers/categorical_column_profile.py:607]:
```suggestion 
        category_count: dict = dict(df_series.value_counts(sort=True).iter_rows())
```
why not 1 line? Does this not run?

**Review by micdavis — APPROVED** [review]

**Review by taylorfturner — APPROVED** [review]

**Pull request merged**